shotover-proxy/src/transforms/cassandra/sink_cluster/token_map.rs

```diff
@@ -0,0 +1,109 @@
+use crate::transforms::cassandra::sink_cluster::CassandraNode;
+use cassandra_protocol::token::Murmur3Token;
+use std::collections::BTreeMap;
+use uuid::Uuid;
+
+#[derive(Debug, Clone)]
+pub struct TokenMap {
+    token_ring: BTreeMap<Murmur3Token, Uuid>,
+}
+
+impl TokenMap {
+    pub fn new(nodes: &[CassandraNode]) -> Self {
+        TokenMap {
+            token_ring: nodes
+                .iter()
+                .flat_map(|node| node.tokens.iter().map(move |token| (*token, node.host_id)))
+                .collect(),
+        }
+    }
+
+    /// Returns local nodes starting at given token and going in the direction of replicas.
+    pub fn nodes_for_token_capped(
```

Comment:
This gives a low level description of the functionality.
Instead we would be better off with a high level description like: `iter_replica_nodes`